### PR TITLE
Home page m UI

### PR DESCRIPTION
**Pull request by komalahire**
Created home page UI

Screenshot from 2022-03-14 19-16-43
Screenshot from 2022-03-14 19-16-52
Screenshot from 2022-03-14 19-16-58
Screenshot from 2022-03-14 19-17-00

**People to loop in / review from**
@jschanker

### CODE REVIEW

File: src/pages/Home/styles.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  },
+
+  cardContent: {
+    flexGrow: 1,
+  },
+
+  engineerPopupCard: {
+    // padding: "10px 45px 5px 0px",
+    width: "253px",
+    background: "red",
```
Comment (jschanker) [inline src/pages/Home/styles.js]: This color style is being overridden, and I assume you didn't want the background of the card to be red?
```suggestion
```

File: src/pages/Home/index.js [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+            <Grid container spacing={4}>
+              <Grid item xs={12} ms={6} md={4}>
+                <Card className={classes.card}>
+                  <CardMedia
+                    className={classes.cardMedia}
+                    component="img"
+                    src={require("./assets/learn-python.svg")}
+                    alt="green iguana"
+                  />
+                  <CardContent className={classes.cardConten}>
```
Comment (jschanker) [inline src/pages/Home/index.js]: Did you mean to use the `cardContent` class here? (There are a number of occurrences.)
```suggestion
                  <CardContent className={classes.cardContent}>
```

File: src/pages/Home/index.js [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+                    </Typography>
+                  </CardContent>
+                </Card>
+              </Grid>
+              <Grid item xs={4}></Grid>
+              <Grid container item xs={4} justifyContent="flex-end">
+                <Card className={classes.engineerPopupCard}>
+                  <Typography align="right" color="text.secondary">
+                    I want to be the first software engineer in my family
+                  </Typography>
```
Comment (jschanker) [inline src/pages/Home/index.js:161]: Why does the first `Card` have `CardContent` and this one not?
```suggestion
                  <CardContent>
                    <Typography align="right" color="text.secondary">
                      I want to be the first software engineer in my family
                    </Typography>
                  </CardContent>
```

File: src/pages/Home/index.js [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+
+          {/* ********************** Section 2 ******************* */}
+
+          <Container sx={{ mt: 4, display: { xs: "none", md: "flex" } }}>
+            <Grid container>
+              <Grid container item xs={4} justifyContent="flex-start">
+                <Card className={classes.typingPopupCard}>
+                  <CardContent>
+                    <Typography color="text.secondary">
+                      I want to be a typing assitant
```
Comment (jschanker) [inline src/pages/Home/index.js]:
```suggestion
                      I want to be a typing assistant
```

File: src/pages/Home/styles.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    marginRight: "60px",
+    padding: "10px 20px 5px 0px",
```
Comment (jschanker) [inline src/pages/Home/styles.js:27]: Once you nest in `<CardContent>`, I don't think you'll need additional padding.
```suggestion
```

File: src/pages/Home/index.js [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+                    </Typography>
+                  </CardContent>
+                </Card>
+              </Grid>
+              <Grid item xs={4}></Grid>
```
Comment (jschanker) [inline src/pages/Home/index.js]: You can avoid creating an unnecessary Grid item by using `justifyContent="space-between"` above. 
```suggestion
```

File: src/pages/Home/index.js [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+                  </Button>
+                </Grid>
+              </Grid>
+            </div>
+          </Container>
+
+          {/* ********************** Section 2 ******************* */}
+
+          <Container sx={{ mt: 4, display: { xs: "none", md: "flex" } }}>
+            <Grid container>
```
Comment (jschanker) [inline src/pages/Home/index.js]: This should create the gap without adding unnecessary HTML via an empty `Grid` item.
```suggestion
            <Grid container justifyContent="space-between">
```

File: src/pages/Home/index.js [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+                  src={require("./assets/Infosys.png")}
+                />
+              </Grid>
+              <Grid item xs={2}>
+                <img
+                  className={classes.partner}
+                  src={require("./assets/Milaap.png")}
+                />
+              </Grid>
+            </Grid>
```
Comment (jschanker) [inline src/pages/Home/index.js]:
```suggestion
            <Grid sx={{ mt: 1 }} container spacing={2}>
              {[
                "Wipro",
                "Infosys",
                "Milaap",
                "Askforhope",
                "Infosys",
                "Milaap"
              ].map((partnerName, _, partnerArr) => (
                <Grid item xs={Math.floor(12 / partnerArr.length)}>
                  <img
                    className={classes.partner}
                    src={require("./assets/" + partnerName + ".png")}
                    alt={partnerName}
                  />
                </Grid>
              ))}
            </Grid>
```

File: src/pages/Home/index.js [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+            </Stack>
+            <Typography sx={{ mt: 3 }} align="center" gutterBottom>
+              Our aim is to help you get technical and soft skills to become
+              ready for advanced learning schools such as Zoho or Navgurukul
+            </Typography>
+          </Container>
+
+          {/* ********************** Section 6 ******************* */}
+
+          <Container maxWidth="md" sx={{ mt: 10 }}>
```
Comment (jschanker) [inline src/pages/Home/index.js:249]: As with functions, you'll generally want to keep Components small and single-task specific so you can potentially reuse them and test/modify them more easily.  As an example, you may want to create a `PartnerSection` component that takes an Array of partner names as a property.  You could then later change this to say a carousel or easily update the list of partners you display by simply modifying the Array.

File: src/pages/Home/index.js [hunk 10]
```diff
@@ -0,0 +1,534 @@
...
+              <Grid item xs={2}>
+                <img
+                  className={classes.partner}
+                  src={require("./assets/Askforhope.png")}
+                />
+              </Grid>
+              <Grid item xs={2}>
+                <img
+                  className={classes.partner}
+                  src={require("./assets/Infosys.png")}
```
Comment (jschanker) [inline src/pages/Home/index.js]: Remember to give images `alt` attributes for accessibility purposes.

**Review by jschanker — COMMENTED** [review]
Here are some initial comments.

File: src/pages/Home/index.js [hunk 11]
```diff
@@ -0,0 +1,534 @@
...
+                    color="primary"
+                  >
+                    <img
+                      className={classes.playstoreImg}
+                      src={require("./assets/playstore.svg")}
+                    />
+                    <span className={classes.downloadBtn}>Download Meraki</span>
+                  </Button>
+                </Grid>
+              </Grid>
```
Comment (jschanker) [inline src/pages/Home/index.js]: Something else that could be turned into its own component, this is duplicated near the bottom.

File: src/pages/Home/index.js (hunk 6 above)
Comment (komalahire) [reply src/pages/Home/index.js]: Ok sure

File: src/pages/Home/index.js (hunk 8 above)
Comment (komalahire) [reply src/pages/Home/index.js]: Yes Done

File: src/pages/Home/index.js (hunk 10 above)
Comment (komalahire) [reply src/pages/Home/index.js]: Ok

File: src/pages/Home/index.js (hunk 9 above)
Comment (komalahire) [reply src/pages/Home/index.js:249]: Yes sure. I will keep in my mind

File: src/pages/Home/index.js (hunk 11 above)
Comment (komalahire) [reply src/pages/Home/index.js]: Sorry. I didn't get it?

Comment (jschanker) [reply src/pages/Home/index.js]: I mean you could create a component like this:
```
function MerakiEntry(props) {
  const isActive = useMediaQuery("(max-width:600px)");
  const classes = useStyles();

  return (
    <div>
      <Typography
        variant="h6"
        color="textPrimary"
        gutterBottom
        {...props.headingAttr}
      >
        With Meraki, begin your programming journey for free today
      </Typography>

      <Grid sx={{ mt: 3 }} container spacing={2} justifyContent="center">
        <Grid item>
          <Button
            className={isActive ? classes.responsiveBtn : classes.LearningBtn}
            variant="contained"
            color="primary"
          >
            Start Learning
          </Button>
        </Grid>
        <Grid item>
          <Button
            className={isActive ? classes.responsiveBtn : classes.LearningBtn}
            variant="outlined"
            color="primary"
          >
            <img
              className={classes.playstoreImg}
              src={require("./assets/playstore.svg")}
              alt="Google Playstore Icon"
            />
            <span className={classes.downloadBtn}>Download Meraki</span>
          </Button>
        </Grid>
      </Grid>
    </div>
  );
}
```
And then lines 35-76 can be replaced by:
```
            <MerakiEntry
              headingAttr={{
                align: isActive ? "left" : "center",
                gutterBottom: true
              }}
            />
```
While lines 448-484 could be replaced by:
```
              <MerakiEntry />
```

File: src/pages/Home/index.js [hunk 12]
```diff
@@ -0,0 +1,534 @@
...
+                      align="center"
+                      component="div"
+                    >
+                      Python
+                    </Typography>
+                    <Typography align="center">
+                      Get familiar with programming with bite sized lessons
+                    </Typography>
+                  </CardContent>
+                </Card>
```
Comment (jschanker) [inline src/pages/Home/index.js]: I'd also recommend making these Components as well so it'd be something like this:
```
function PathwayCard(props) {
  const classes = useStyles();
  return (
    <Card className={classes.card}>
      <CardMedia
        component="img"
        src={require("./assets/" + props.imgSrc + ".svg")}
        alt={props.alt || "image for " + props.trackName}
      />
      <CardContent className={classes.cardContent}>
        <Typography gutterBottom variant="h6" align="center" component="div">
          {props.trackName}
        </Typography>
        <Typography align="center">{props.description}</Typography>
      </CardContent>
    </Card>
  );
}
```
And then you could make some sort of `PathwayTable` component rendering an Array of `PathwayCard` components.  But I'm assuming this breaking up into components will happen naturally once you starting pulling this information from the database.  (See also: https://reactjs.org/docs/thinking-in-react.html#step-1-break-the-ui-into-a-component-hierarchy.)

File: src/pages/Home/index.js (hunk 11 above)
Comment (komalahire) [reply src/pages/Home/index.js]: Done

File: src/pages/Home/index.js (hunk 12 above)
Comment (komalahire) [reply src/pages/Home/index.js]: Ok sure

**Review by Poonam-Singh-Bagh — APPROVED** [review]

**Review by Poonam-Singh-Bagh — APPROVED** [review]

**Pull request merged**